addons/website_event_crm/models/event_registration.py

```diff
@@ -24,3 +24,12 @@ def _get_lead_description_fields(self):
         res = super(EventRegistration, self)._get_lead_description_fields()
         res.append('registration_answer_ids')
         return res
+
+    def _get_lead_values(self, rule):
+        """Update lead values from Lead Generation rules to include the visitor and their language"""
+        lead_values = super()._get_lead_values(rule)
+        new_lead_values = {'visitor_ids': self.visitor_id}
+        if self.visitor_id.website_id.language_count > 1:
+            new_lead_values['lang_id'] = self.visitor_id.lang_id.id
```

Comment:
Is this still necessary? Can't we always take the visitor language?

Reply:
We could, but I feel like it's misleading for the user.
It would mean that even if the website is available in only one language (e.g. Arabic), then all leads created through event registrations will get the Arabic language assigned.
So when trying to contact those leads, we might use the wrong language.

Reply:
Not sure if it's misleading. If the website is only available in one language, I think that we can assume that users using it can speak that language. It could even specifically be the goal here to target only people who speak it. IMO we could always take the visitor language.

Reply:
@jeh-odoo Okay, I removed it